### DISCUSSION TITLE
Homekey with Ethernet (wired)

### DISCUSSION THREAD

**Discussion by Miker2022** [rednblkx/HomeKey-ESP32#97] [General]
Hello,

I really like this project, it perfectly fits in my environment.

However, my WiFi-signal is very weak at the enrtance door and the ESP has to sit in a metal case which makes things even worse. 

My question: is there an easy way to run this on a Raspberry PI using wired Ethernet or is there any other solution? I am searching for ESP32-Ethernet for quite a while but did not find any detailed information.

Regards
M.

**Comment by rednblkx** [comment]
Hello,

Ethernet support is something i wish to add sometime in the future. 

For Raspberry Pi, you can look at the [proof-of-concept](https://github.com/kormax/apple-home-key-reader) written in Python by @kormax who originally reverse-engineered the homekey, but you will have to make some modifications in order to make use of it, don't know if i can help with that as i haven't used it for this purpose.

**Reply by Schinken6** [reply]
Can't I just buy something like this and everything works? https://shop.everythingsmart.io/products/est-poe-32-esp32-poe-development-board

**Reply by rednblkx** [reply]
Sure, if you have the right software to go along with it, however, this project currently does not support Ethernet (or PoE if you will)

**Reply by 011V32** [reply]
Looks like things are going in the right direction now.
https://github.com/HomeSpan/HomeSpan/discussions/738#discussioncomment-11552123

**Reply by Miker2022** [reply]
Homespan 2.1.0 now supports wired Ethernet. Will Homekey use that Feature in the near Future?

Michael

**Reply by rednblkx** [reply]
It’s in the works https://github.com/rednblkx/HomeKey-ESP32/pull/167

You can try the latest CI build if you want https://github.com/rednblkx/HomeKey-ESP32/actions/runs/12772805385

**Comment by electric-m** [comment]
Ethernet Support would be awesome.
I have mine working, but have it disconnected regularly due to bad WiFi signal, even though I have repeater on the other side of the wall.
The existing case of the doorbell is made of metal…

**Comment by GhostingBit** [comment]
+1 for Ethernet support, I also like to hard wire everything up.

but as a workaround the ESPHome component should do the trick, as ESPHome has Ethernet support, and ESPHome can also be used with MQTT, so you can replicate functionality more or less.

**Reply by Miker2022** [reply]
…could you please explain the trick a bit further? 

How do you establish network connectivity from the ESP running the project Software?

**Reply by GhostingBit** [reply]
Have a look at the devs other repo:
https://github.com/rednblkx/HAP-ESPHome/

You flags ESPHome with the devs HAP-ESPHome component which contains the lock entity. And you set up ESPHome with a wired LAN port rather than WiFi. This should work just fine, until maybe to this project wired Ethernet is being added by a contributor or maintainer. 

@rednblkx do normal NFC (eg Mifare) cards also work with the HAP-ESPHome component? Would need to be able to get the NFC cards ID.

**Comment by cerahmed** [comment]
Came here for the same question. 

So are you all saying that running this library on an esp32-poe would not work?
Appreciate if someone could clarify.

**Reply by dlbogdan** [reply]
Not clear anywhere if the protocol still works (unlocks the door) if the wifi is down or unreachable. Is it still working ?

**Reply by ianchuter** [reply]
I have this project running on an ESP-32 POE2 from Olimex. Ethernet works fine as long as I don't rely on the POE power - yet to fault find that.
